docs/concepts/json_schema.md

```diff
@@ -617,6 +617,66 @@ print(json.dumps(Model.model_json_schema(), indent=2))
 """
 ```
 
+#### Merging `json_schema_extra`
+
+Starting in v2.9, Pydantic merges `json_schema_extra` dictionaries from annotated types:
```

Comment:
I think you should add some discussion of why we made this change to try to pacify anyone angry about a "breaking" change. I think it's arguably a bug fix if it's increasing consistency, though depends on the details I guess.